src/gt4sd/properties/molecules/core.py

```diff
@@ -0,0 +1,238 @@
+    Nature chemistry, 4(2), 90-98.
+    """
+    return _qed(to_mol(mol))
+
+
+def number_of_h_acceptors(mol: SmallMolecule) -> int:
+    """Calculate number of H acceptors of a molecule."""
+    return num_H_acceptors(to_mol(mol))
+
+
+def number_of_atoms(mol: SmallMolecule) -> int:
+    """Calculate number of atoms of a molecule."""
+    return num_atoms(to_mol(mol))
+
+
+def number_of_h_donors(mol: SmallMolecule) -> int:
+    """Calculate number of H donors of a molecule."""
+    return num_H_donors(to_mol(mol))
+
+
+def number_of_aromatic_rings(mol: SmallMolecule) -> int:
+    """Calculate number of aromatic rings of a molecule."""
+    return num_aromatic_rings(to_mol(mol))
+
+
+def number_of_rings(mol: SmallMolecule) -> int:
+    """Calculate number of rings of a molecule."""
+    return num_rings(to_mol(mol))
+
+
+def number_of_rotatable_bonds(mol: SmallMolecule) -> int:
+    """Calculate number of rotatable bonds of a molecule."""
+    return num_rotatable_bonds(to_mol(mol))
+
+
+def number_of_large_rings(mol: SmallMolecule) -> int:
+    """Calculate the amount of large rings (> 6 atoms) of a molecule."""
+    ringinfo = to_mol(mol).GetRingInfo()
+    return len([x for x in ringinfo.AtomRings() if len(x) > 6])
+
+
+def molecular_weight(mol: SmallMolecule) -> float:
+    """Calculate molecular weight of a molecule."""
+    return mol_weight(to_mol(mol))
+
+
+def is_scaffold(mol: SmallMolecule) -> int:
+    """Whether a molecule is identical to its Murcko Scaffold."""
+    mol = to_mol(mol)
+    smi = Chem.MolToSmiles(mol, canonical=True)
+    return int(smi == MurckoScaffoldSmiles(mol=mol))
+
+
+def number_of_heterocycles(mol: SmallMolecule) -> int:
+    """The amount of heterocycles of a molecule."""
+    return CalcNumHeterocycles(to_mol(mol))
+
+
+def number_of_stereocenters(mol: SmallMolecule) -> int:
+    """The amount of stereo centers of a molecule."""
+    return CalcNumAtomStereoCenters(to_mol(mol))
+
+
+def similarity_to_seed(
+    mol: SmallMolecule, similarity_fn: Callable[[SmallMolecule], Property]
+) -> Property:
+    """Calculate the similarity of a molecule to a seed molecule.
+
+    For example:
+    ```py
+    from gt4sd.properties.molecules import similarity_to_seed, get_similarity_fn
+
+    func = get_similarity_fn(target_mol='CCO', fp_key='FCFP4')
+    similarity_to_seed(mol='CCC', similarity_fn=func)
+    ```
+
+    """
+    return similarity_fn(to_smiles(mol))
+
+
+def activity_against_target(
+    mol: SmallMolecule, affinity_fn: Callable[[SmallMolecule], Property]
+) -> Property:
+    """Calculate the activity of a molecule against a target molecule.
+
+    For example:
+    ```py
+    from gt4sd.properties.molecules import activity_against_target, get_activity_fn
+
+    func = get_activity_fn(target_mol='DRD2')
+    activity_against_target(mol='CCC', affinity_fn=func)
+    ```
+
+    """
+    return affinity_fn(to_smiles(mol))
+
+
+# TODO: Need to put the trained models on COS and implement caching logic
```

Comment:
This is something that should be relatively easy if we introduce a dedicated sub-bucket path.